src/components/dashboard/data/utils.js

```diff
@@ -8,7 +37,7 @@ import { ASSIGNMENT_TYPES } from '../../enterprise-user-subsidy/enterprise-offer
  */
 export default function getActiveAssignments(assignments = []) {
   const activeAssignments = assignments.filter((assignment) => [
-    ASSIGNMENT_TYPES.CANCELLED, ASSIGNMENT_TYPES.ALLOCATED,
+    ASSIGNMENT_TYPES.ALLOCATED,
   ].includes(assignment.state));
```

Comment:
@adamstankiewicz any concerns with changing this definition of `getActiveAssignments()`?

Reply:
@iloveagent57 @katrinan029 Yes, by removing `ASSIGNMENT_TYPES.CANCELLED` here, the `CourseEnrollments` components that check against "cancelled" assignment states would no longer be provided any canceled assignments to process (i.e., I don't believe the util functions (e.g., `getIsActiveCancelledAssignment`) would be executed? 🤔 For example, trying to run this locally, I couldn't get the canceled alert to appear unless canceled assignments were considered here.

I might recommend modifying `getActiveAssignments` to split out arrays for each assignment lifecycle state that's relevant, which would enable more granular control over which types of assignments you're working with in `CourseEnrollments`, i.e.:

```jsx
// Renamed `getActiveAssignments` to `getAssignmentsByState`

export function getAssignmentsByState(assignments = []) {
  const allAssignments = [];
  const allocatedAssignments = [];
  const canceledAssignments = [];
  const acceptedAssignments = [];

  assignments.forEach((assignment) => {
    allAssignments.push(assignment);
    if (assignment.state === ASSIGNMENT_TYPES.ALLOCATED) {
      allocatedAssignments.push(assignment);
    }
    if (assignment.state === ASSIGNMENT_TYPES.CANCELLED) {
      canceledAssignments.push(assignment);
    }
    if (assignment.state === ASSIGNMENT_TYPES.ACCEPTED) {
      acceptedAssignments.push(assignment);
    }
  });

  const hasAssignments = allAssignments.length > 0;
  const hasAllocatedAssignments = allocatedAssignments.length > 0;
  const hasCanceledAssignments = canceledAssignments.length > 0;
  const hasAcceptedAssignments = acceptedAssignments.length > 0;

  return {
    assignments,
    hasAssignments,
    allocatedAssignments,
    hasAllocatedAssignments,
    canceledAssignments,
    hasCanceledAssignments,
    acceptedAssignments,
    hasAcceptedAssignments,
};
```

Note: This function should really only be called within `getRedeemablePoliciesData` at this point (as it is on `master`) as that logic has already been executed further up in the component tree, without needing to run again to get similar results.